File: HomeAutomation/src/Components/MotionDetector.java
package Components;

import Core_Models.IObserver;
import Core_Models.SystemState;

public class MotionDetector implements IObserver {
    public boolean motionDetected;
    public long lastDetectedTime;

    public void update(SystemState state) {
        this.motionDetected = state.motionDetected;
        if (motionDetected)
        {
            lastDetectedTime = System.currentTimeMillis();
            logMotion();
        }
        else {
            System.out.println("Motion not detected.");
        }
    }

    private void logMotion() {
        System.out.println("Motion detected at: " + lastDetectedTime);
    }

    public boolean isMotionDetected(){
        return motionDetected;
    }

}

File: HomeAutomation/HomeAutomation/HomeAutomation/MotionDetection/MotionDetector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HomeAutomation
{
    public class MotionDetector:IObserver
    {

        public List<string> detectionLogs = new List<string>();
        public void Notify(EnvironmentalState state)
        {
            if (state.isMotionDetected)
            {
                var LogDetected = $"Motion detected at {DateTime.Now} in {state.Location}";
                detectionLogs.Add(LogDetected);
                Console.WriteLine(LogDetected);
            }
            else
            {
                Console.WriteLine("No motion detected.");
            }
        }

    }
}
